src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java

```diff
@@ -116,18 +120,22 @@
 
     protected final MessageDigest messageDigest;
 
+    protected final CredentialStoreReference credentialStoreReference;
```

Comment:
I think we can still go ahead and get this merged as it is mostly internal stuff but I don't think the mechanism should need to be CredentialStore aware - instead the mechanism should be using callbacks to either obtain the clear text representation of the password or the pre-hashed representation of the password and the callback handler then handling the conversion between the credential store reference and the callback handler.

That way the conversion between the two concepts is centralised into a single point so all existing and new mechanisms (including third party mechanisms) get this integration for free.